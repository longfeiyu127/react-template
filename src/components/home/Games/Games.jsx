import React from 'react';
import { Tabs, Badge } from 'antd-mobile';
import PropTypes from 'prop-types';
import GameItem from './GameItem/GameItem';
import './Games.less';
import games from '../../../service/config/games.json';

const tabs = [
  { title: <Badge>全部</Badge> },
  { title: <Badge>热门</Badge> },
  // { title: <Badge text={'今日(20)'}>Second Tab</Badge> },
  { title: <Badge>分类</Badge> }
];

export default class HomeGames extends React.Component {
  static propTypes = {
    history: PropTypes.object.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    // console.log(games);
    // console.log(this.props.history)
    const { history } = this.props;
    const AllGames = games.games.map(item => (
      /* eslint-disable */
      <GameItem
        itemData={item}
        history={history}
        img={require(`../../../assets/images/games/${item.icon}`)}
        key={item.path}
      />
      /* eslint-enable */
    ));
    return (
      <div className="home-games">
        <Tabs
          tabs={tabs}
          initialPage={0}
          onChange={(tab, index) => {
            console.log('onChange', index, tab);
          }}
          onTabClick={(tab, index) => {
            console.log('onTabClick', index, tab);
          }}
        >
          <div>{AllGames}</div>
          <div>{AllGames}</div>
          <div>{AllGames}</div>
        </Tabs>
      </div>
    );
  }
}
