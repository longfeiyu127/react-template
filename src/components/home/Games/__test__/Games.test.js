import React from 'react';
import renderer from 'react-test-renderer';
import Games from '../Games';

describe('<Games />', () => {
  it('component render', () => {
    let games = renderer.create(<Games history={{}} />).toJSON();
    expect(games).toMatchSnapshot();

    games = renderer.create(<Games history={{}} />).toJSON();
    expect(games).toMatchSnapshot();
  });
});
