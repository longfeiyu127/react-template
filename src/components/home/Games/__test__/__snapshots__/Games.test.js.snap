// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Games /> component render 1`] = `
<div
  className="home-games"
>
  <div
    className="am-tabs am-tabs-horizontal am-tabs-top"
  >
    <div
      className="am-tabs-tab-bar-wrap"
    >
      <div
        className="am-tabs-default-bar am-tabs-default-bar-animated am-tabs-default-bar-top"
        style={
          Object {
            "backgroundColor": "#fff",
          }
        }
      >
        <div
          className="am-tabs-default-bar-content"
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tablist"
          style={
            Object {
              "touchAction": "pan-y",
            }
          }
        >
          <div
            aria-selected={true}
            className="am-tabs-default-bar-tab am-tabs-default-bar-tab-active"
            id="m-tabs-0-0"
            onClick={[Function]}
            role="tab"
            style={
              Object {
                "width": "33.333333333333336%",
              }
            }
          >
            <span
              className="am-badge"
            >
              全部
            </span>
          </div>
          <div
            aria-selected={false}
            className="am-tabs-default-bar-tab"
            id="m-tabs-0-1"
            onClick={[Function]}
            role="tab"
            style={
              Object {
                "width": "33.333333333333336%",
              }
            }
          >
            <span
              className="am-badge"
            >
              热门
            </span>
          </div>
          <div
            aria-selected={false}
            className="am-tabs-default-bar-tab"
            id="m-tabs-0-2"
            onClick={[Function]}
            role="tab"
            style={
              Object {
                "width": "33.333333333333336%",
              }
            }
          >
            <span
              className="am-badge"
            >
              分类
            </span>
          </div>
          <div
            className="am-tabs-default-bar-underline"
            style={
              Object {
                "left": "0%",
                "width": "33.333333333333336%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="am-tabs-content-wrap am-tabs-content-wrap-animated"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "MozTransform": "translate3d(0%, 0px, 1px)",
          "WebkitTransform": "translate3d(0%, 0px, 1px)",
          "touchAction": "pan-x pan-y",
          "transform": "translate3d(0%, 0px, 1px)",
        }
      }
    >
      <div
        aria-hidden={false}
        aria-labelledby="m-tabs-0-0"
        className="am-tabs-pane-wrap am-tabs-pane-wrap-active"
        role="tabpanel"
        style={Object {}}
      >
        <div>
          <div
            className="home-TabItem"
          >
            <img
              alt="icon"
              className="TabItem-banner"
              src="lightning_icon.png"
            />
            <div
              className="TabItem-content"
            >
              <h5
                className="title"
              >
                单机雷电
              </h5>
              <p>
                点击雷电是一款炫酷华丽的飞机射击类单机游戏！超华丽的画面、超炫酷的爆炸效果、超流畅的手感，让你在万里苍穹一展雄姿！攻击特效华丽夸张！刺激火爆的场面
              </p>
            </div>
            <div
              className="TabItem-button"
              onClick={[Function]}
            >
              打开
            </div>
          </div>
          <div
            className="home-TabItem"
          >
            <img
              alt="icon"
              className="TabItem-banner"
              src="tictactoe_launcher.png"
            />
            <div
              className="TabItem-content"
            >
              <h5
                className="title"
              >
                井字棋
              </h5>
              <p>
                井字棋游戏是一款非常好的亲子游戏,既锻炼小孩子的思维,又能培养亲子之间的感情,来吧!一起体验井字棋的乐趣吧!
              </p>
            </div>
            <div
              className="TabItem-button"
              onClick={[Function]}
            >
              打开
            </div>
          </div>
          <div
            className="home-TabItem"
          >
            <img
              alt="icon"
              className="TabItem-banner"
              src="fivechess_icon.png"
            />
            <div
              className="TabItem-content"
            >
              <h5
                className="title"
              >
                五子棋
              </h5>
              <p>
                五子棋游戏的历史可谓源远流长,是一款老少皆宜的两人对弈纯策略游戏,讲究的是有攻有守的五子棋技巧,玩法简单易上手,五个同色棋子连成一线就可以获胜
              </p>
            </div>
            <div
              className="TabItem-button"
              onClick={[Function]}
            >
              打开
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        aria-labelledby="m-tabs-0-1"
        className="am-tabs-pane-wrap am-tabs-pane-wrap-inactive"
        role="tabpanel"
        style={Object {}}
      >
        <div>
          <div
            className="home-TabItem"
          >
            <img
              alt="icon"
              className="TabItem-banner"
              src="lightning_icon.png"
            />
            <div
              className="TabItem-content"
            >
              <h5
                className="title"
              >
                单机雷电
              </h5>
              <p>
                点击雷电是一款炫酷华丽的飞机射击类单机游戏！超华丽的画面、超炫酷的爆炸效果、超流畅的手感，让你在万里苍穹一展雄姿！攻击特效华丽夸张！刺激火爆的场面
              </p>
            </div>
            <div
              className="TabItem-button"
              onClick={[Function]}
            >
              打开
            </div>
          </div>
          <div
            className="home-TabItem"
          >
            <img
              alt="icon"
              className="TabItem-banner"
              src="tictactoe_launcher.png"
            />
            <div
              className="TabItem-content"
            >
              <h5
                className="title"
              >
                井字棋
              </h5>
              <p>
                井字棋游戏是一款非常好的亲子游戏,既锻炼小孩子的思维,又能培养亲子之间的感情,来吧!一起体验井字棋的乐趣吧!
              </p>
            </div>
            <div
              className="TabItem-button"
              onClick={[Function]}
            >
              打开
            </div>
          </div>
          <div
            className="home-TabItem"
          >
            <img
              alt="icon"
              className="TabItem-banner"
              src="fivechess_icon.png"
            />
            <div
              className="TabItem-content"
            >
              <h5
                className="title"
              >
                五子棋
              </h5>
              <p>
                五子棋游戏的历史可谓源远流长,是一款老少皆宜的两人对弈纯策略游戏,讲究的是有攻有守的五子棋技巧,玩法简单易上手,五个同色棋子连成一线就可以获胜
              </p>
            </div>
            <div
              className="TabItem-button"
              onClick={[Function]}
            >
              打开
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        aria-labelledby="m-tabs-0-2"
        className="am-tabs-pane-wrap am-tabs-pane-wrap-inactive"
        role="tabpanel"
        style={Object {}}
      />
    </div>
  </div>
</div>
`;

exports[`<Games /> component render 2`] = `
<div
  className="home-games"
>
  <div
    className="am-tabs am-tabs-horizontal am-tabs-top"
  >
    <div
      className="am-tabs-tab-bar-wrap"
    >
      <div
        className="am-tabs-default-bar am-tabs-default-bar-animated am-tabs-default-bar-top"
        style={
          Object {
            "backgroundColor": "#fff",
          }
        }
      >
        <div
          className="am-tabs-default-bar-content"
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tablist"
          style={
            Object {
              "touchAction": "pan-y",
            }
          }
        >
          <div
            aria-selected={true}
            className="am-tabs-default-bar-tab am-tabs-default-bar-tab-active"
            id="m-tabs-1-0"
            onClick={[Function]}
            role="tab"
            style={
              Object {
                "width": "33.333333333333336%",
              }
            }
          >
            <span
              className="am-badge"
            >
              全部
            </span>
          </div>
          <div
            aria-selected={false}
            className="am-tabs-default-bar-tab"
            id="m-tabs-1-1"
            onClick={[Function]}
            role="tab"
            style={
              Object {
                "width": "33.333333333333336%",
              }
            }
          >
            <span
              className="am-badge"
            >
              热门
            </span>
          </div>
          <div
            aria-selected={false}
            className="am-tabs-default-bar-tab"
            id="m-tabs-1-2"
            onClick={[Function]}
            role="tab"
            style={
              Object {
                "width": "33.333333333333336%",
              }
            }
          >
            <span
              className="am-badge"
            >
              分类
            </span>
          </div>
          <div
            className="am-tabs-default-bar-underline"
            style={
              Object {
                "left": "0%",
                "width": "33.333333333333336%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="am-tabs-content-wrap am-tabs-content-wrap-animated"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "MozTransform": "translate3d(0%, 0px, 1px)",
          "WebkitTransform": "translate3d(0%, 0px, 1px)",
          "touchAction": "pan-x pan-y",
          "transform": "translate3d(0%, 0px, 1px)",
        }
      }
    >
      <div
        aria-hidden={false}
        aria-labelledby="m-tabs-1-0"
        className="am-tabs-pane-wrap am-tabs-pane-wrap-active"
        role="tabpanel"
        style={Object {}}
      >
        <div>
          <div
            className="home-TabItem"
          >
            <img
              alt="icon"
              className="TabItem-banner"
              src="lightning_icon.png"
            />
            <div
              className="TabItem-content"
            >
              <h5
                className="title"
              >
                单机雷电
              </h5>
              <p>
                点击雷电是一款炫酷华丽的飞机射击类单机游戏！超华丽的画面、超炫酷的爆炸效果、超流畅的手感，让你在万里苍穹一展雄姿！攻击特效华丽夸张！刺激火爆的场面
              </p>
            </div>
            <div
              className="TabItem-button"
              onClick={[Function]}
            >
              打开
            </div>
          </div>
          <div
            className="home-TabItem"
          >
            <img
              alt="icon"
              className="TabItem-banner"
              src="tictactoe_launcher.png"
            />
            <div
              className="TabItem-content"
            >
              <h5
                className="title"
              >
                井字棋
              </h5>
              <p>
                井字棋游戏是一款非常好的亲子游戏,既锻炼小孩子的思维,又能培养亲子之间的感情,来吧!一起体验井字棋的乐趣吧!
              </p>
            </div>
            <div
              className="TabItem-button"
              onClick={[Function]}
            >
              打开
            </div>
          </div>
          <div
            className="home-TabItem"
          >
            <img
              alt="icon"
              className="TabItem-banner"
              src="fivechess_icon.png"
            />
            <div
              className="TabItem-content"
            >
              <h5
                className="title"
              >
                五子棋
              </h5>
              <p>
                五子棋游戏的历史可谓源远流长,是一款老少皆宜的两人对弈纯策略游戏,讲究的是有攻有守的五子棋技巧,玩法简单易上手,五个同色棋子连成一线就可以获胜
              </p>
            </div>
            <div
              className="TabItem-button"
              onClick={[Function]}
            >
              打开
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        aria-labelledby="m-tabs-1-1"
        className="am-tabs-pane-wrap am-tabs-pane-wrap-inactive"
        role="tabpanel"
        style={Object {}}
      >
        <div>
          <div
            className="home-TabItem"
          >
            <img
              alt="icon"
              className="TabItem-banner"
              src="lightning_icon.png"
            />
            <div
              className="TabItem-content"
            >
              <h5
                className="title"
              >
                单机雷电
              </h5>
              <p>
                点击雷电是一款炫酷华丽的飞机射击类单机游戏！超华丽的画面、超炫酷的爆炸效果、超流畅的手感，让你在万里苍穹一展雄姿！攻击特效华丽夸张！刺激火爆的场面
              </p>
            </div>
            <div
              className="TabItem-button"
              onClick={[Function]}
            >
              打开
            </div>
          </div>
          <div
            className="home-TabItem"
          >
            <img
              alt="icon"
              className="TabItem-banner"
              src="tictactoe_launcher.png"
            />
            <div
              className="TabItem-content"
            >
              <h5
                className="title"
              >
                井字棋
              </h5>
              <p>
                井字棋游戏是一款非常好的亲子游戏,既锻炼小孩子的思维,又能培养亲子之间的感情,来吧!一起体验井字棋的乐趣吧!
              </p>
            </div>
            <div
              className="TabItem-button"
              onClick={[Function]}
            >
              打开
            </div>
          </div>
          <div
            className="home-TabItem"
          >
            <img
              alt="icon"
              className="TabItem-banner"
              src="fivechess_icon.png"
            />
            <div
              className="TabItem-content"
            >
              <h5
                className="title"
              >
                五子棋
              </h5>
              <p>
                五子棋游戏的历史可谓源远流长,是一款老少皆宜的两人对弈纯策略游戏,讲究的是有攻有守的五子棋技巧,玩法简单易上手,五个同色棋子连成一线就可以获胜
              </p>
            </div>
            <div
              className="TabItem-button"
              onClick={[Function]}
            >
              打开
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        aria-labelledby="m-tabs-1-2"
        className="am-tabs-pane-wrap am-tabs-pane-wrap-inactive"
        role="tabpanel"
        style={Object {}}
      />
    </div>
  </div>
</div>
`;
